### PR TITLE
wip: Remove Faraday Middleware

### PR DESCRIPTION
**Pull request by bf4**

### CODE REVIEW

File: lib/qbo_api/connection.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 require 'faraday'
-require 'faraday_middleware'
+require 'faraday/response/json'
+require 'qbo_api/raise_http_error'
```
Comment (minimul) [inline lib/qbo_api/connection.rb:3]: `require 'qbo_api/raise_http_exception'` ?

File: lib/qbo_api/connection.rb [hunk 2]
```diff
@@ -68,11 +70,7 @@ def response(resp, entity: nil)
     end
 
     def parse_response_body(resp)
-      body = resp.body
-      case resp.headers['Content-Type']
-      when /json/ then JSON.parse(body)
-      else body
-      end
+      resp.body
```
Comment (minimul) [inline lib/qbo_api/connection.rb:73]: Nice!

File: lib/qbo_api/raise_http_exception.rb [hunk 3]
```diff
@@ -1,32 +1,43 @@
 require 'faraday'
 require 'nokogiri'
 # @private
-module FaradayMiddleware
+module QboApi
```
Comment (minimul) [inline lib/qbo_api/raise_http_exception.rb:4]: That name spacing pattern with the gem is:
```rb
class QboApi
   module Name
  ...
....
```

**Review by minimul — CHANGES_REQUESTED** [review]
Thanks @bf4 for taking this on!

**Comment by Greyoxide** [comment]
Are there any updates on this?  I'd love to use this awesome gem but the Faraday mis-match is messing me up.

**Comment by bf4** [comment]
Personally if me I need to run the test suite etc. I just wrote this
quickly to show the way and then forgot about it,  oops

I think there might also be a behavior change around mime type parsing
which may come up.

**Comment by minimul** [comment]
It is not hitting any pain points for me at the moment but I'll merge in quality fix and re-release quickly if you all or someone wants to tackle it.

**Comment by Greyoxide** [comment]
I'd love to help but I know so little about the gem or the faraday_middleware gem, I'm worried I'd break something.  If someone wanted to outline specific areas to focus on and what needed to happen, I'd be willing to take a crack at it.

**Comment by bf4** [comment]
The change needed is basically this pr.  I'm not around to test if the
changed are good or not and ci didn't run.  But I've replaced all my other
faraday json usages in my app which used faraday_middleware with a change
like in the pr.

**Comment by bf4** [comment]
Superseded by https://github.com/minimul/qbo_api/pull/118

File: lib/qbo_api/raise_http_exception.rb (hunk 3 above)
Comment (bf4) [reply lib/qbo_api/raise_http_exception.rb:4]: 🤦

File: qbo_api.gemspec [hunk 4]
```diff
@@ -25,8 +25,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'dotenv'
   spec.add_development_dependency 'vcr'
   spec.add_development_dependency 'amazing_print'
-  spec.add_runtime_dependency 'faraday', '~> 1.10.0'
-  spec.add_runtime_dependency 'faraday_middleware'
+  spec.add_runtime_dependency 'faraday', '>= 1.10.0'
```
Comment (bf4) [inline qbo_api.gemspec:28]: intention here is that gem can, for now, remain compatible with both 1.x and 2x. per my observations

**Pull request closed without merging**